midi2/src/utility.rs

```diff
@@ -97,15 +97,59 @@ mod timestamp {
 mod delta_clockstamp {
     use crate::detail::common_properties;
     use crate::utility;
+    use crate::ux::{u20, u4};
     pub const STATUS: u8 = 0b0100;
     #[midi2_proc::generate_message(Via(crate::utility::Utility), FixedSize, MinSizeUmp(1))]
     struct DeltaClockstamp {
         #[property(common_properties::UmpMessageTypeProperty<{utility::UMP_MESSAGE_TYPE}>)]
         ump_type: (),
         #[property(common_properties::ChannelVoiceStatusProperty<STATUS>)]
         status: (),
-        #[property(utility::DataProperty)]
-        time_data: u16,
+        #[property(DataProperty)]
+        time_data: u20,
+    }
+
+    struct DataProperty;
+
+    impl<B: crate::buffer::Ump> crate::detail::property::Property<B> for DataProperty {
+        type Type = u20;
+    }
+
+    impl<'a, B: crate::buffer::Ump> crate::detail::property::ReadProperty<'a, B> for DataProperty {
+        fn read(buffer: &'a B) -> Self::Type {
+            use crate::detail::BitOps;
+            use core::ops::BitOr;
+            let bits: u20 = buffer.buffer()[0].nibble(3).into();
+            let bits = bits << 16;
+            bits.bitor(u20::new(buffer.buffer()[0].word(1) as u32))
```

Comment:
Rather than piecing the value together from separate sections, it would be simpler to to just construct it in one go, using one big mask. e.g:

```rust
u20::new(buffer.buffer()[0] & 0x000F_FFFF)
```

This would also avoid the need to cast back and forth between all these different numeric types.